File: examples/primitives/python/wrappers/enums/color_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "cxx/enums/color.hpp"

namespace py = pybind11;

void bindColor(py::module& m) {
    py::enum_<iegen::example::Color>(m, "Color")
        .value("Red", iegen::example::Color::Red)
        .value("Green", iegen::example::Color::Green)
        .value("Blue", iegen::example::Color::Blue)
        .export_values();
}

File: examples/primitives/python/src/enums/color_pygen.py
from __future__ import annotations
from typing import *
import examples.enums.color as pybind_color_pygen
from bind_utils import *
from metaclass import *


class Color(pybind_color_pygen.Color):
    """
    Color Enum."""
    # Red = 1
    Red = pybind_color_pygen.Color.Red
    # Green = 1
    Green = pybind_color_pygen.Color.Green
    # Blue = 20
    Blue = pybind_color_pygen.Color.Blue
